src/support/err.c

```diff
@@ -554,21 +554,21 @@ __wt_panic_func(WT_SESSION_IMPL *session, int error, const char *func, int line,
         WT_PANIC, func, line, category, WT_VERBOSE_ERROR, "the process must exit and restart", ap));
     va_end(ap);
 
-    if (EXTRA_DIAGNOSTICS_ENABLED(session, WT_DIAG_PANIC)) {
-        /*
-         * In the diagnostic builds, we want to drop core in case of panics that are not due to data
-         * corruption. A core could be useful in debugging.
-         *
-         * In the case of corruption, we want to be able to test the application's capability to
-         * salvage by returning an error code. But we do not want to lose the ability to drop core
-         * if required. Hence in the diagnostic mode, the application can set the debug flag to
-         * choose between dropping a core and returning an error.
-         */
-        if (conn != NULL &&
-          (!F_ISSET(conn, WT_CONN_DATA_CORRUPTION) ||
-            FLD_ISSET(conn->debug_flags, WT_CONN_DEBUG_CORRUPTION_ABORT)))
-            __wt_abort(session);
-    }
+#ifdef HAVE_DIAGNOSTIC
```

Comment:
I've moved this back to `#ifdef HAVE_DIAGNOSTIC` for now. 
It'd be useful to dump cores in this case, but there's some assumptions that we're running in test mode that I want to untangle before having this runtime enable-able